include/host-build.mk

```diff
@@ -165,7 +165,15 @@ ifndef DUMP
 		touch $$@
 
   $(call Host/Exports,$(HOST_STAMP_INSTALLED))
-  $(HOST_STAMP_INSTALLED): $(HOST_STAMP_BUILT) $(if $(FORCE_HOST_INSTALL),FORCE)
+  ifeq ($(CONFIG_IN_SDK),y)
+    # if we're in SDK, don't re-install host tools if the stamp already exists,
+    # as newly compiled host tools might not be compatible with the SDK libs
+    $(HOST_STAMP_INSTALLED): \
+      $(if $(wildcard $(HOST_STAMP_INSTALLED)),$$(warning $(PKG_NAME) host part is already built, to force re-compilation remove $(HOST_STAMP_INSTALLED)),$(HOST_STAMP_BUILT)) \
+      $(if $(FORCE_HOST_INSTALL),FORCE)
+  else
+    $(HOST_STAMP_INSTALLED): $(HOST_STAMP_BUILT) $(if $(FORCE_HOST_INSTALL),FORCE)
+  endif
```

Comment:
Hi @mpratt14, (replying to https://github.com/openwrt/openwrt/pull/10255#issuecomment-1190903797)

Good catch! I don't use GNU Make often, so I forgot that multiple rules for one target combine prerequisites. How's the following changes?

I've moved the `FORCE_HOST_INSTALL` line to be next to the recipe at the bottom, since I feel that looks slightly clearer.

```suggestion
  ifeq ($(CONFIG_IN_SDK),y)
    # if we're in SDK, don't re-install host tools if the stamp already exists,
    # as newly compiled host tools might not be compatible with the SDK libs
    $(HOST_STAMP_INSTALLED): $(if $(wildcard $(HOST_STAMP_INSTALLED)),$$(warning $(PKG_NAME) host part is already built, to force re-compilation remove $(HOST_STAMP_INSTALLED)),$(HOST_STAMP_BUILT))
  else
    $(HOST_STAMP_INSTALLED): $(HOST_STAMP_BUILT)
  endif
  $(HOST_STAMP_INSTALLED): $(if $(FORCE_HOST_INSTALL),FORCE)
```

It might even be worth completely removing the `FORCE_HOST_INSTALL` bit when we're in the SDK, as re-installing SDK bundled binaries would fail, since they're not compiled. If we do compile them (e.g. by using `$(if $(FORCE_HOST_INSTALL),FORCE $(HOST_STAMP_BUILT))`, they'll be using the host libraries, not the SDK host libraries, so might fail in the SDK (see #7299).

Reply:
yeah looks better

I dont know anything about how FORCE_HOST_INSTALL is used, but if it's not for the SDK it could be recombined with the else statement prerequisite

Reply:
> I dont know anything about how FORCE_HOST_INSTALL is used, but if it's not for the SDK it could be recombined with the else statement prerequisite

The only thing I could find in this SDK was: https://github.com/openwrt/openwrt/commit/050902efd70bf8d4c88c8ce4124bd8cbe15548c7, it's used for running [scripts/deptest.sh](https://github.com/openwrt/openwrt/tree/ec6293febc244d187e71a6e54f44920be679cde4/scripts/deptest.sh).

However, I also found [this old OpenWRT package setting `FORCE_HOST_INSTALL:=y`](https://github.com/hellocccat/Archer-C2600-GPL/blob/e102814f50a4b249f970e949b333c092099aff57/openwrt/toolchain/kernel-headers/Makefile#L26).

It's possible that some other 3rd-party OpenWRT packages set `FORCE_HOST_INSTALL:=y`, and my GitHub/grep.app search couldn't find them. Because of this, my gut feeling is to keep backwards compatibility in the SDK when `FORCE_HOST_INSTALL=y` (last two lines changed from https://github.com/openwrt/openwrt/pull/10255#discussion_r926754176).

```suggestion
  ifeq ($(CONFIG_IN_SDK),y)
    # if we're in SDK, don't re-install host tools if the stamp already exists,
    # as newly compiled host tools might not be compatible with the SDK libs
    $(HOST_STAMP_INSTALLED): $(if $(wildcard $(HOST_STAMP_INSTALLED)),$$(warning $(PKG_NAME) host part is already built, to force re-compilation remove $(HOST_STAMP_INSTALLED)),$(HOST_STAMP_BUILT))
  else
    $(HOST_STAMP_INSTALLED): $(HOST_STAMP_BUILT)
  endif
  # set HOST_STAMP_BUILT if FORCE_HOST_INSTALL, since it might be missing in SDK
  $(HOST_STAMP_INSTALLED): $(if $(FORCE_HOST_INSTALL),FORCE $(HOST_STAMP_BUILT))
```

If you're okay with this, I'll include these changes, and rebase & force-push my commit.

Reply:
@Ansuel  what do you think?